### PR TITLE
Migrate to github actions

### PR DESCRIPTION
**Pull request by thoov**
Moving off of travis-ci / appveyor and onto github ci

### CODE REVIEW

File: .github/workflows/nodejs.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+name: CI
+
+on: [push]
```
Comment (rwjblue) [inline .github/workflows/nodejs.yml]: You also must use `pull_request`. If you don't folks that PR this repo from their own forks will not run CI.

My suggested setup:

```suggestion
on:
  pull_request:
  push:
    # filtering branches here, prevents duplicate builds from pull_request and push
    branches: 
      - master
      - 'v*'
    # always run CI for tags
    tags:
      - '*'
      
  # early issue detection, run CI weekly on Sundays  
  schedule:
  - cron:  '0 6 * * 0' 
```

File: .github/workflows/nodejs.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+jobs:
+  test:
+    name: Test on node ${{ matrix.node_version }} and ${{ matrix.os }}
```
Comment (rwjblue) [inline .github/workflows/nodejs.yml]: The long name here will make the CI status's pretty hard to read.

Screen Shot 2019-10-02 at 9 43 23 AM

If you make this `Tests` only, the CI status's _will_ still include the dynamic bits automatically.

```suggestion
    name: Tests
```

File: .github/workflows/nodejs.yml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    - name: Use Node.js ${{ matrix.node_version }}
+      uses: actions/setup-node@v1
+      with:
+        version: ${{ matrix.node_version }}
+  
+    - name:  install yarn
+      run: npm install -g yarn
+
+    - name: install dependencies
+      run: yarn install
```
Comment (rwjblue) [inline .github/workflows/nodejs.yml]:
```suggestion
      run: yarn install --frozen-lockfile
```

**Comment by thoov** [comment]
cc: @rwjblue @stefanpenner @oligriffiths 

This should be good to go now (just squash the commits down). We will also need to turn off the appveyor integration once this lands.

**Pull request merged**